### PR TITLE
Basic polling query alignment

### PR DESCRIPTION
**Pull request by Poincare**
Introduces a really simple method of aligning polling queries. In most applications, there are may be queries that operate on exactly the same interval. The scheduler will fire these on the same interval meaning that the queries will be batched automatically by the batcher. Solves #401.

TODO:

- [ ] Update CHANGELOG.md with your change
- [ ] Make sure all of the significant new logic is covered by tests
- [ ] Rebase your changes on master so that they can be merged easily
- [ ] Make sure all tests and linter rules pass

### CODE REVIEW

**Comment by Poincare** [comment]
@stubailo ready for review

File: src/scheduler.ts [hunk 1]
```diff
@@ -55,51 +65,100 @@ export class QueryScheduler {
...
+    queryId?: string,
+    firstFetch: boolean = true,
+    listener?: QueryListener
+  ): string {
+    if (!options.pollInterval) {
+      throw new Error('Attempted to start a polling query without a polling interval.');
+    }
+
     if (!queryId) {
       queryId = this.queryManager.generateQueryId();
```
Comment (stubailo) [inline src/scheduler.ts]: This line isn't tested!

File: src/QueryManager.ts [hunk 2]
```diff
@@ -96,23 +96,29 @@ export class ObservableQuery extends Observable<ApolloQueryResult> {
...
-
+    const queryManager = scheduler.queryManager;
     const queryId = queryManager.generateQueryId();
+    const isPollingQuery = !!options.pollInterval;
+
     const subscriberFunction = (observer: Observer<ApolloQueryResult>) => {
       const retQuerySubscription = {
         unsubscribe: () => {
+          if (isPollingQuery) {
+            scheduler.stopPollingQuery(queryId);
```
Comment (stubailo) [inline src/QueryManager.ts]: Looks like we don't have any tests for this condition.

**Comment by Poincare** [comment]
@stubailo ready for review

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,11 +3,31 @@
 Expect active development and potentially significant breaking changes in the `0.x` track. We'll try to be diligent about releasing a `1.0` version in a timely fashion (ideally within 3 to 6 months), to signal the start of a more stable API.
 
 ### vNEXT
+<<<<<<< HEAD
```
Comment (stubailo) [inline CHANGELOG.md]: It looks like we have committed a merge conflict

Comment (Poincare) [inline CHANGELOG.md]: Fixed

File: src/QueryManager.ts [hunk 4]
```diff
@@ -468,18 +468,11 @@ export class QueryManager {
 
   public startQuery(queryId: string, options: WatchQueryOptions, listener: QueryListener) {
     this.queryListeners[queryId] = listener;
-    this.fetchQuery(queryId, options);
 
-    if (options.pollInterval) {
-      if (options.noFetch) {
-        throw new Error('noFetch option should not use query polling.');
```
Comment (stubailo) [inline src/QueryManager.ts]: @Poincare we lost this code in the merge, it seems

Comment (Poincare) [inline src/QueryManager.ts]: Ok will fix

**Pull request merged**